### PR TITLE
AO3-4109 Making next/previous work links in series more accessible

### PR DESCRIPTION
**Pull request by pixelred**
## Issue

https://otwarchive.atlassian.net/browse/AO3-4109

## Purpose

changes `»` link to next work in series to `Next Work →`
changes `«` link to previous work in series to `← Previous Work`

### CODE REVIEW

**Comment by sarken** [comment]
It looks like there are some tests that are looking for the old style links, so they'll need to be updated to look for the new "Next Work" and "Previous Work" links instead. (You can find out which files need updating by looking at [the build on Travis](https://travis-ci.org/otwcode/otwarchive/builds/179364577), clicking on the things that are red/failing, and then scrolling down to the red `(::) failed steps (::)` section. It'll tell you exactly what lines and files to look at in the /features folder.)

I'm also not sure if we want to make the new link text into translatable strings -- in theory we want to, but we're in a weird place regarding translation at the moment, so I will ask around and get back to you!

**Comment by pixelred** [comment]
will do! 👍

File: app/helpers/series_helper.rb [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
     series.map do |serial|
       serial_works = serial.serial_works.find(:all, :include => :work, :conditions => ['works.posted = ?', true], :order => :position).select{|sw| sw.work.visible(current_user)}.collect{|sw| sw.work}
-      visible_position = serial_works.index(work) || serial_works.length     
+      visible_position = serial_works.index(work) || serial_works.length
       unless !visible_position
-        previous_link = (visible_position > 0) ? link_to("&laquo; ".html_safe, serial_works[visible_position - 1]) : "".html_safe
-        main_link = ("Part " + (visible_position+1).to_s + " of the " + link_to(serial.title, serial) + " series").html_safe
-        next_link = (visible_position < serial_works.size-1) ? link_to(" &raquo;".html_safe, serial_works[visible_position + 1]) : "".html_safe
+        previous_link = (visible_position > 0) ? link_to("&larr; Previous Work".html_safe, serial_works[visible_position - 1]) : "".html_safe
+        main_link = (" Part " + (visible_position+1).to_s + " of the " + link_to(serial.title, serial) + " series ").html_safe
```
Comment (ariana-paris) [inline app/helpers/series_helper.rb]: This raw mix of strings and variables arranged in English-language order will be impossible to translate when the day comes, particularly without being wrapped in `ts`. While you're editing this line, I'd suggest using interpolation to make it a single string, e.g.:

```ruby
main_link = ts("Part #{(visible_position + 1).to_s} of the #{link_to(serial.title, serial)} series").html_safe
```

or alternatively to make the text itself more legible:

```ruby
main_link = ts("Part %{position} of the %{series_title} series", position: (visible_position + 1).to_s, series_title: link_to(serial.title, serial)).html_safe
```

The spaces at either end feel as though they've been added for aesthetic reasons that should probably be addressed in the aesthetic layer? (i.e. HTML or CSS)

File: app/helpers/series_helper.rb [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
         previous_link + main_link + next_link
       end
     end
   end
-  
+
   def work_series_description(work, series)
     serial = SerialWork.where(:work_id => work.id, :series_id => series.id).first
-    ("Part <strong>#{serial.position}</strong> of " + link_to(series.title, series)).html_safe 
+    ("Part <strong>#{serial.position}</strong> of " + link_to(series.title, series)).html_safe
```
Comment (ariana-paris) [inline app/helpers/series_helper.rb]: I know this isn't actually part of the changes, but it should also be wrapped in `ts` and interpolated:

```ruby
ts("Part <strong>#{serial.position}</strong> of #{link_to(series.title, series)}").html_safe
```

You know, while you're in there...

File: app/helpers/series_helper.rb [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
     series = work.series.select{|s| s.visible?(current_user)}
     series.map do |serial|
       serial_works = serial.serial_works.find(:all, :include => :work, :conditions => ['works.posted = ?', true], :order => :position).select{|sw| sw.work.visible(current_user)}.collect{|sw| sw.work}
-      visible_position = serial_works.index(work) || serial_works.length     
+      visible_position = serial_works.index(work) || serial_works.length
       unless !visible_position
-        previous_link = (visible_position > 0) ? link_to("&laquo; ".html_safe, serial_works[visible_position - 1]) : "".html_safe
-        main_link = ("Part " + (visible_position+1).to_s + " of the " + link_to(serial.title, serial) + " series").html_safe
-        next_link = (visible_position < serial_works.size-1) ? link_to(" &raquo;".html_safe, serial_works[visible_position + 1]) : "".html_safe
+        previous_link = (visible_position > 0) ? link_to("&larr; Previous Work".html_safe, serial_works[visible_position - 1]) : "".html_safe
```
Comment (ariana-paris) [inline app/helpers/series_helper.rb]: The text in the `link_to` should be wrapped in the `ts` method which some day will allow us to track down all translatable strings:

```ruby
previous_link = visible_position > 0 ? link_to(ts("&larr; Previous Work").html_safe, serial_works[visible_position - 1]) : "".html_safe
```

File: app/helpers/series_helper.rb [hunk 4]
```diff
@@ -1,28 +1,28 @@
...
+        main_link = (" Part " + (visible_position+1).to_s + " of the " + link_to(serial.title, serial) + " series ").html_safe
+        next_link = (visible_position < serial_works.size-1) ? link_to("Next Work &rarr;".html_safe, serial_works[visible_position + 1]) : "".html_safe
```
Comment (ariana-paris) [inline app/helpers/series_helper.rb]: So nice to say goodbye to the fiddly unclickable guillemets! 👍

File: app/helpers/series_helper.rb (hunk 2 above)
Comment (pixelred) [inline app/helpers/series_helper.rb]: will do :' )

**Comment by pixelred** [comment]
travis... 😞

i'll try this again

**Comment by sarken** [comment]
Alas, this has merge conflicts that will need to be fixed before it can be merged. No rush, though!

**Comment by pixelred** [comment]
did it work? 👀

**Comment by sarken** [comment]
It's mergeable now, but it looks like things got a little mixed up -- the changes to the tests are totally gone, so they're going to fail when they run (they're looking for the old style series links now, not the new code you wrote).

**Comment by pixelred** [comment]
sorry about that; i think i got it right this time? hopefully

**Comment by sarken** [comment]
I think that's exactly what's going on here, yeah -- AO3-4786 significantly changed series.feature, so instead of trying to merge two vastly different files, it might be easiest to copy the contents of the new test files, commit that, and then re-do whatever test changes are needed to cover the new navigation format.

**Comment by pixelred** [comment]
i went and copied the test from that branch over and then changed the little arrows in the tests to the new series navigation links, and the test failed but it was for comments and kudos, so i think the series tests now cover the changes in link text

**Comment by sarken** [comment]
I've restarted the comment and kudos part, but it looks good to me! 👍

**Pull request merged**